src/epoch/ut1.rs

```diff
@@ -83,8 +83,37 @@ impl Epoch {
     }
 }
 
+#[cfg_attr(feature = "python", pymethods)]
+impl Epoch {
+
+    #[staticmethod]
+    #[pyo3(name = "from_ut1_duration")]
+    pub fn py_from_ut1_duration(duration: Duration, provider: PyRef<Ut1Provider>) -> PyResult<Self> {
+        Ok(Epoch::from_ut1_duration(duration, &*provider))
+    }
+
+    #[pyo3(name = "ut1_offset")]
+    pub fn py_ut1_offset(&self, provider: PyRef<Ut1Provider>) -> Option<Duration> {
+        self.ut1_offset(&*provider)
+    }
+
+    #[pyo3(name = "to_ut1_duration")]
+    pub fn py_to_ut1_duration(&self, provider: PyRef<Ut1Provider>) -> Duration {
+        self.to_ut1_duration(&*provider)
+    }
+
+    #[pyo3(name = "to_ut1")]
+    pub fn py_to_ut1(&self, provider: PyRef<Ut1Provider>) -> Self {
+        self.to_ut1(&*provider)
+    }
+}
+
 #[cfg_attr(kani, derive(kani::Arbitrary))]
-#[derive(Copy, Clone, Debug, Default, Tabled)]
+#[cfg_attr(feature = "python",
+    pyo3::pyclass(module = "hifitime", name = "DeltaTaiUt1", get_all)
+)]
+#[cfg_attr(not(feature = "python"), derive(Copy))]  // Copy only when NOT exposing to Python
```

Comment:
Why only copy when not exposing to Python? Is the copy an issue? We recently improved the performance of the Ut1 provider in #419.